### PR TITLE
CSEN-22 "TagRequest" classes were implemented

### PR DESCRIPTION
**Pull request by MedvedevBW**

### CODE REVIEW

**Comment by MedvedevBW** [comment]
@placccebo

File: src/main/scala/com/bwsw/cloudstack/entities/requests/TagCreateRequest.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    i = i + 1
+  }
+}
+
+object TagCreateRequest {
+  /**
+    * Class is responsible for providing tags creation settings.
+    *
+    * @param resourceType type of creating tags
+    * @param resourceIds a set with UUID of resources within which tags will be created
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/requests/TagCreateRequest.scala]: a set of resource's UUIDs for which tags will be created

File: src/main/scala/com/bwsw/cloudstack/entities/requests/TagFindRequest.scala [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+import java.util.UUID
+
+import br.com.autonomiccs.apacheCloudStack.client.ApacheCloudStackRequest
+import com.bwsw.cloudstack.entities.requests.traits.Request
+import com.bwsw.cloudstack.entities.requests.util.Constants.Commands
+import com.bwsw.cloudstack.entities.requests.util.Constants.Parameters._
+import com.bwsw.cloudstack.entities.requests.util.traits.TagType
+
+class TagFindRequest extends Request {
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/requests/TagFindRequest.scala]: add a description

File: src/main/scala/com/bwsw/cloudstack/entities/requests/util/UserTagType.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package com.bwsw.cloudstack.entities.requests.util
+
+import com.bwsw.cloudstack.entities.requests.util.traits.TagType
+
+class UserTagType extends TagType {
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/requests/util/UserTagType.scala]: add a description

File: src/main/scala/com/bwsw/cloudstack/entities/requests/util/traits/TagType.scala [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package com.bwsw.cloudstack.entities.requests.util.traits
+
+trait TagType {
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/requests/util/traits/TagType.scala]: add a description

File: src/test/scala/com/bwsw/cloudstack/entities/requests/TagCreateRequestTestSuite.scala [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    new ApacheCloudStackApiCommandParameter(RESPONSE,"json"),
+    new ApacheCloudStackApiCommandParameter(RESOURCE_TYPE, tagType.toString),
+    new ApacheCloudStackApiCommandParameter(RESOURCE_IDS, resourceIds.mkString(",")),
+    new ApacheCloudStackApiCommandParameter("tags[0].key", key1),
+    new ApacheCloudStackApiCommandParameter("tags[0].value", value1),
+    new ApacheCloudStackApiCommandParameter("tags[1].key", key2),
+    new ApacheCloudStackApiCommandParameter("tags[1].value", value2)
+  )
+
+  it should "create a request with predefined parameters" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/entities/requests/TagCreateRequestTestSuite.scala]: not only predefined parameters

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/main/scala/com/bwsw/cloudstack/entities/requests/TagCreateRequest.scala (hunk 1 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/TagCreateRequest.scala]: ok

File: src/main/scala/com/bwsw/cloudstack/entities/requests/TagFindRequest.scala (hunk 2 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/TagFindRequest.scala]: ok

File: src/main/scala/com/bwsw/cloudstack/entities/requests/util/traits/TagType.scala (hunk 4 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/util/traits/TagType.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/entities/requests/TagCreateRequestTestSuite.scala (hunk 5 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/entities/requests/TagCreateRequestTestSuite.scala]: ok

**Review by placccebo — APPROVED** [review]

**Pull request merged**